### PR TITLE
Add @InternalApi annotations and private markers on internal apis

### LINKED ISSUE #22506: make stream impl internal
I think we should opt for making as much as possible in impl InternalApi. A later design task will be to make it possible to plugin custom phases or materializers.

### PR DESCRIPTION
**Pull request by 2m**
Fixes #22506

### CODE REVIEW

**Comment by patriknw** [comment]
One trick I sometimes use is to run `sbt doc` and look at the scaladoc. The `impl` package should not show up, unless we have some good reason for making them public already. Try that.

**Comment by 2m** [comment]
Great tip. I needed to add a few more `private[akka]` in order to have `.impl` package gone from the scala doc.

File: akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala [hunk 1]
```diff
@@ -44,18 +45,18 @@ final case class GraphStageModule[+S <: Shape @uncheckedVariance, +M](
 /**
  * INTERNAL API
  */
-object GraphStages {
+@InternalApi private[akka] object GraphStages {
```
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: Is the private/internal api below really needed when the surrounding scope is already internal api/private[akka]?

**Review by johanandren — APPROVED** [review]
LGTM, great tedious work @2m !

File: akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala (hunk 1 above)
Comment (2m) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: Yes. You are right. This is an artifact of the find and replace. But it does not hurt to have it below as well. Then you do not need context to see that these are internal apis when browsing the source code.

Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: Yeap, no problem, just made me wonder.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**